ansibledriver/service/ansible.py

```diff
@@ -221,127 +228,280 @@ def __init__(self, ansible_properties, request_id, lifecycle, display=None):
 
-    def v2_playbook_on_vars_prompt(self, varname, private=True, prompt=None, encrypt=None, confirm=False, salt_size=None, salt=None, default=None, unsafe=None):
-        logger.debug('v2_playbook_on_vars_prompt {0}'.format(varname))
-
-    def v2_runner_item_on_ok(self, result):
-        logger.debug('v2_runner_item_on_ok {0}'.format(result))
+    def _log_unreachable_event(self, result):
+        if self.ansible_properties.log_progress_events:
+            task_name = result._task.get_name()
+            self._clean_results(result._result, result._task.action)
+            task_result = result._result
+            event = HostUnreachableEvent(task_name=task_name, host_name=result._host.get_name().strip(), task_result=task_result)
+            delegated_vars = result._result.get('_ansible_delegated_vars', None)
+            if delegated_vars is not None:
+                event.delegated_host_name = delegated_vars['ansible_host']
+            self.event_logger.add(event)
 
-    def v2_runner_item_on_failed(self, result):
-        logger.debug('v2_runner_item_on_failed {0}'.format(result))
+    def __handle_unreachable(self, result):
+        # TODO do not overwrite if already set
+        self.failed_task = result._task.get_name()
+        self.host_unreachable_log.append(dict(task=self.failed_task, result=result._result))
+        self.host_unreachable = True
+        self.failure_reason = 'Resource unreachable (task ' + str(self.failed_task) + ' failed: ' + str(result._result) + ')'
+        self.failure_details = FailureDetails(FAILURE_CODE_RESOURCE_NOT_FOUND, self.failure_reason)
+        self.playbook_failed = True
+        self._log_unreachable_event(result)
 
-    def v2_runner_item_on_skipped(self, result):
-        logger.debug('v2_runner_item_on_skipped {0}'.format(result))
+    def v2_playbook_on_vars_prompt(self, varname, private=True, prompt=None, encrypt=None, confirm=False, salt_size=None, salt=None, default=None, unsafe=None):
+        """
+        Called when a var_prompt is used in a playbook, which we can't support because the playbook is not running in an interactive shell
+        """
+        logger.debug('v2_playbook_on_vars_prompt: {0}'.format(varname))
+        if self.ansible_properties.log_progress_events:
+            event = VarPromptEvent(var_name=varname)
+            self.event_logger.add(event)
 
     def runner_on_no_hosts(self):
         logger.debug('runner_on_no_hosts')
 
     def v2_runner_retry(self, result):
-        logger.debug('v2_runner_retry {0}'.format(result))
+        """
+        Called when a task is retried
+        """
+        logger.debug('v2_runner_retry: {0}'.format(result))
+        if self.ansible_properties.log_progress_events:
+            host_name = result._host.get_name().strip()
+            delegated_vars = result._result.get('_ansible_delegated_vars', None)
+            if delegated_vars is not None:
```

Comment:
I assume result._host and result._result are always != None (I think they are)?

Reply:
I believe so, from what I can see in their code base.